fairing/ml_tasks/tasks.py

```diff
@@ -64,8 +65,18 @@ def create(self):
         self.url = self._deployer.deploy(self.pod_spec)
         logger.warning("Prediction endpoint: {}".format(self.url))
 
-    def predict(self, data):
-        r = requests.post(self.url, data=data)
+    def predict_nparray(self, data, names=["t:0", "t:1"]):
+        pdata={
+            "data": {
+                "names":names,
```

Comment:
names should be a 1D string array, so doing some check here would be useful.

Reply:
I don't think its expected that we do type validation everywhere

Reply:
The issue here is that if there is an error, it will be in the serving container logs which is hard to get to. From customer UX perspective, doing a small validation here goes a long way in avoiding a laborious debugging.

Reply:
we print out the server error, which should tell the user the same info

Reply:
The error I am getting if the names filed is something like this. I think this is fine for now but definitely can be improved with client side validation
```
'{"status":{"code":-1,"info":"Invalid JSON: Failed to parse names field: expected string or bytes-like object.","reason":"MICROSERVICE_BAD_DATA","status":1}}\n'
```